### PR TITLE
feat(lytfs): allow use of env var to auth requests

### LINKED ISSUE #1: [Feature] Allow use of PATs for authenticated requests OR use raw.githubusercontent.com
### Description

I was trying to implement some form of automation for
l-zeuch/yagpdb.vim and ran into the following error:

```
fetching base context functions: fetching common/templates/context.go
from GitHub: GET
https://api.github.com/repos/botlabs-gg/yagpdb/contents/common/templates/context.go?ref=master:
403 API rate limit exceeded for [IP redacted]. (But here's the good
news: Authenticated requests get a higher rate limit. Check out the
documentation for more details.) [rate reset in 25m57s] 
```

It'd be great if we could provide either a flag or an environment
variable to pass along a GitHub PAT, so that we can avoid this error.

Arguably this is more or less user error, as this program is not
intended to be ran multiple times in quick succession, however exactly
for such development moments it'd be great to have such an option.

For reference, the library you're using to interact with GitHub has
support for this, prominently placed in the README:
https://github.com/google/go-github#authentication

Do note that I'm willing to implement this with an appropriate patch,
however it is kind of rude to just do it and not ask first.

----

Alternatively I think it is sufficient to just use
`raw.githubusercontent.com` instead of `api.github.com` — this is
generally the way install scripts and similar are also shared, if hosted
on GitHub, as that does not require authentication nor is it rate
limited as much as the API.

### PR DESCRIPTION
**Pull request by l-zeuch**
This is a combination of two commits:
- chore(deps): upgrade google/go-github to v55
- feat(lytfs): allow use of env var to auth requests

chore(deps:) upgrade google/go-github to v55

  Upgrade google/go-github to v55 (latest), so that the WithAuthToken
  configurator becomes available for us to use.
  
  Additionally this will get rid of the transient dependency on
  "golang.org/x/auth/oauth2", but that is tangential.
  
  Signed-off-By: Luca Zeuch <l-zeuch@email.de>

feat(lytfs): allow use of env var to auth requests

  Allow passing along an environment variable to authenticate requests to
  the GitHub API so that clients have a higher ratelimit.
  
  This is useful for developers wanting to use yagfuncdata, as often
  enough it is just not possible to do so without rapid testing.
  
  Closes #1
  
  Signed-off-by Luca Zeuch <l-zeuch@email.de>

### CODE REVIEW

File: cmd/lytfs/main.go [hunk 1]
```diff
@@ -17,7 +17,9 @@ var (
 func usage() {
 	fmt.Fprintln(os.Stderr, `lytfs: list available YAGPDB template function names
 	
-usage: lytfs [-timeout duration]`)
+usage: lytfs [-timeout duration]
+
+To authenticate your requests, pass a GitHub personal access token via the YAGFUNCDATA_AUTH_TOKEN environment variable.`)
```
Comment (l-zeuch) [inline cmd/lytfs/main.go]: This phrasing is obviously up for debate, but it gets the point across.

**Review by jo3-l — CHANGES_REQUESTED** [review]
Having `NewGithubClient` change behavior depending on the presence of an environment variable is surprising IMO; that should be done by the program that uses `yagfuncdata`. So WDYT about leaving `DefaultFileContentProvider` as is and instead changing `lytfs` to call `NewGitHubFileProvider` with an authenticated GitHub client if needed?

File: cmd/lytfs/main.go [hunk 2]
```diff
@@ -28,7 +31,13 @@ func main() {
...
 	defer cancel()
 
-	sources := yagfuncdata.DefaultSources(yagfuncdata.DefaultFileContentProvider)
+	var sources []yagfuncdata.Source
+	if token := os.Getenv("YAGFUNCDATA_AUTH_TOKEN"); token != "" {
+		fcp := yagfuncdata.NewGitHubFileProvider(github.NewClient(nil).WithAuthToken(token), "botlabs-gg", "yagpdb", "master")
+		sources = yagfuncdata.DefaultSources(fcp)
+	} else {
+		sources = yagfuncdata.DefaultSources(yagfuncdata.DefaultFileContentProvider)
+	}
```
Comment (jo3-l) [inline cmd/lytfs/main.go]: This is a nit, but I would rather write
```suggestion
	fcp := yagfuncdata.DefaultFileContentProvider
	if token := os.Getenv("YAGFUNCDATA_AUTH_TOKEN"); token != "" {
		fcp = yagfuncdata.NewGitHubFileProvider(github.NewClient(nil).WithAuthToken(token), "botlabs-gg", "yagpdb", "master")
	}
	sources := yagfuncdata.DefaultSources(fcp)
```

Comment (l-zeuch) [reply cmd/lytfs/main.go]: Sure, that reads cleaner anyway.

**Review by jo3-l — APPROVED** [review]
LGTM, thanks for your patience :-)

**Comment by l-zeuch** [comment]
Thanks, no worries :D

**Pull request merged**